File: Can Place Flowers.py
from typing import List

class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        placed = 0
        for index, element in enumerate(flowerbed):
            right_invalid, left_invalid = None, None
            if element == 1:
                continue
            if index + 1 == len(flowerbed):
                right_invalid = False
            else:
                right_invalid = flowerbed[index+1] == 1
            if index > 0:
                left_invalid = flowerbed[index-1] == 1
            else:
                left_invalid = False

            if right_invalid or left_invalid:
                continue
            else:
                flowerbed[index] = 1
                placed += 1
        return placed >= n

File: cpp/Can Place Flowers.cpp
#include <string>
#include <iostream>
#include <numeric>
#include <vector>
using namespace std;

class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int flowersplaced = 0;
        for (int i=0; i< flowerbed.size(); i++) {
            if (i==0) {
                if (flowerbed.size() == 1) {
                    if (flowerbed[i] == 0) {
                        flowersplaced++;
                        flowerbed[i] = 1;
                    }
                } else if (flowerbed[i] == 0 && flowerbed[i+1] == 0) {
                    flowersplaced++;
                    flowerbed[i] = 1;
                }
            } else if (i==flowerbed.size()-1) {
                if (flowerbed[i]==0 && flowerbed[i-1]==0) {
                    flowersplaced++;
                    flowerbed[i] = 1;
                }
            } else if (flowerbed[i] == 0 && flowerbed[i-1]==0 && flowerbed[i+1] == 0) {
                flowersplaced += 1;
                flowerbed[i] = 1;
            }
        }
        return flowersplaced >= n;
    }
};